### ISSUE TITLE
Distributed Tracing configuration does not work

### ISSUE THREAD

**Issue by kgrozdanovski** [getsentry/sentry-docs#4653]
### Core or SDK?

Platform/SDK

### Which part? Which one?

Symfony

### Description

After following the steps and using the code snippets for configuring Distributed Tracing with a sampler function, I get the following error:
```shell
Please add the class to service &quot;sentry_tracer_sampler&quot; even if it is constructed by a factory since we might need to add method calls based on compile-time checks. (500 Internal Server Error)
```

My assumption is that the example for the `sentry.yaml` configuration is incorrect, however I am not sure what the correct settings should be.

### Suggested Solution

The following code snippet from the documentation needs to be updated with a functional configuration:

```yaml
sentry:
    options:
        traces_sampler: 'sentry_traces_sampler'

services:
    sentry_traces_sampler:
        factory: ['@\App\Service\Sentry', 'getTracesSampler']
```

**Comment by imatwawana** [comment]
hey @stayallive - can you take a look, please?

**Comment by kgrozdanovski** [comment]
I just got it to work using an example provided in this [comment](https://github.com/getsentry/sentry-symfony/issues/336#issuecomment-746633997).

Here is the proper, working services configuration in `sentry.yaml`:
```yaml
services:
    App\Service\Sentry:
        autowire: true
    sentry_tracer_sampler:
        autowire: true
        factory: ['App\Service\Sentry', 'getTracesSampler']
        class: 'App\Service\Sentry'
```
And the matching options configuration in the same file:
```yaml
 options:
        traces_sampler: 'sentry_tracer_sampler'
```

**Comment by stayallive** [comment]
@kgrozdanovski you are correct, this is going to be addressed with: #4658.

Thanks for pointing this out 👍
